design/one-pager-release-process.md

```diff
@@ -0,0 +1,172 @@
+# Release Process and Engineering
+
+## Background
+
+The current lifecycle process for a new version of the Crossplane software has gotten the project through the `v0.3` release, but also has a number of shortcomings.
+A summary of the current process that is very succinct, and skips many of the details, would look like the following:
+
+1. The team creates a [milestone](https://github.com/crossplaneio/crossplane/milestones) to represent the current/active release
```

Comment:
For this OSS project, it might help to be more specific about the team members involved here.

"maintainer team"? "maintainers"?